sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/PemKeySource.java

```diff
@@ -91,4 +104,46 @@ private RSAPublicKey extractRsaPublicKey(X509Certificate certificate)
     }
     return publicKey;
   }
+
+  private boolean isPublicKey(URI pemKeyLocation) {
+    try (InputStream is = pemKeyLocation.toURL().openStream()) {
+      String keyContent = readContent(is).trim();
+      return keyContent.startsWith("-----BEGIN PUBLIC KEY-----")
+          && keyContent.endsWith("-----END PUBLIC KEY-----");
+    } catch (IOException e) {
+      throw new KeyLoadFailedException("Failed to read key from " + pemKeyLocation, e);
+    }
+  }
+
+  private RSAPublicKey loadPublicKey(URI pemKeyLocation) {
+    LOG.info("Loading public key for token signature verification from PEM {}", pemKeyLocation);
+    try (InputStream is = pemKeyLocation.toURL().openStream()) {
+      String pemPublicKeyContent = readContent(is);
+      String publicKeyPem =
+          pemPublicKeyContent
+              .replace("-----BEGIN PUBLIC KEY-----", "")
+              .replaceAll("(\\r\\n|\\r|\\n)", "")
+              .replace("-----END PUBLIC KEY-----", "");
+
+      byte[] encoded = Base64.decodeBase64(publicKeyPem);
+      X509EncodedKeySpec keySpec = new X509EncodedKeySpec(encoded);
+      final KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+      return (RSAPublicKey) keyFactory.generatePublic(keySpec);
+    } catch (ClassCastException
+        | IOException
+        | NoSuchAlgorithmException
+        | InvalidKeySpecException e) {
+      throw new KeyLoadFailedException("Failed to load public key at " + pemKeyLocation, e);
+    }
+  }
+
+  private String readContent(InputStream content) throws IOException {
```

Comment:
What's our policy of just using `IOUtils#toString` for these basic operations?

Reply:
Apache Commons IO is not in the classpath of the Auth module and I'm a bit wary with Guava as they introduce breaking changes quite often. This pattern is quite common unless we fully upgrade to Java 11.

Reply:
I also wouldn't go with Guava. Ok, then I'll approve it :)